app/controllers/api/v1/listen_signals_controller.rb

```diff
@@ -18,21 +18,21 @@ def update
   end
 
   def create
-    brand = current_user.brand
-    identity = brand.identities.first
-    signal_params = signal_params(params)
-    signal_params[:brand] = brand
-    signal_params[:identity] = identity
-    @listen_signal = ListenSignal.create!(signal_params)
-
-    default_response_msg = params[:default_response]
-    repeat_response_msg = params[:repeat_response]
-    create_grouped_response(default_response_msg, repeat_response_msg)
+    signal_params            = signal_params(params)
+    signal_params[:brand]    = @brand
+    signal_params[:identity] = @brand.twitter_identity
+    default_response_msg     = params[:default_response]
+    repeat_response_msg      = params[:repeat_response]
+
+    ActiveRecord::Base.transaction do
+      @listen_signal = ListenSignal.create!(signal_params)
+      create_grouped_response(default_response_msg, repeat_response_msg)
+    end
 
     if @listen_signal
```

Comment:
The`if-else` statement is unnecessary because you are using`#create!`. You can just render.